src/modules/mavlink/streams/BATTERY_STATUS_V2.hpp

```diff
@@ -0,0 +1,137 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef BATTERY_STATUS_V2_HPP
+#define BATTERY_STATUS_V2_HPP
+
+#include <uORB/topics/battery_status.h>
+
+class MavlinkStreamBatteryStatusV2 : public MavlinkStream
+{
+public:
+	static MavlinkStream *new_instance(Mavlink *mavlink) { return new MavlinkStreamBatteryStatusV2(mavlink); }
+
+	static constexpr const char *get_name_static() { return "BATTERY_STATUS_V2"; }
+	static constexpr uint16_t get_id_static() { return MAVLINK_MSG_ID_BATTERY_STATUS_V2; }
+
+	const char *get_name() const override { return get_name_static(); }
+	uint16_t get_id() override { return get_id_static(); }
+
+	unsigned get_size() override
+	{
+		static constexpr unsigned size_per_battery = MAVLINK_MSG_ID_BATTERY_STATUS_V2_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
+		return size_per_battery * _battery_status_subs.advertised_count();
+	}
+
+private:
+	explicit MavlinkStreamBatteryStatusV2(Mavlink *mavlink) : MavlinkStream(mavlink) {}
+
+	uORB::SubscriptionMultiArray<battery_status_s, battery_status_s::MAX_INSTANCES> _battery_status_subs{ORB_ID::battery_status};
+
+	bool send() override
+	{
+		bool updated = false;
+
+		for (auto &battery_sub : _battery_status_subs) {
+			battery_status_s battery_status;
+
+			if (battery_sub.update(&battery_status)) {
+				mavlink_battery_status_v2_t bat_msg{};
+
+				bat_msg.id = battery_status.id;
+				bat_msg.temperature = (battery_status.connected && PX4_ISFINITE(battery_status.temperature)) ?
+						      static_cast<int16_t>(battery_status.temperature * 1e2f) : INT16_MAX;
+				bat_msg.voltage = (battery_status.connected
+						   && PX4_ISFINITE(battery_status.voltage_v)) ? battery_status.voltage_v : float(NAN);
```

Comment:
This check is mostly good, but will output 0 if `BatteryStatus` contains `0`, and it should be outputting NaN in that case.

The reason is that [BatteryStatus.voltage](https://docs.px4.io/main/en/msg_docs/BatteryStatus.html#batterystatus-uorb-message) indicates the PX4 invalid value for voltage is 0 (and we need to output NaN for invalid values - see  [BATTERY_STATUS_V2.voltage](https://mavlink.io/en/messages/development.html#BATTERY_STATUS_V2).

Can you recheck the other values against the documented PX4 uorb invalid values.